cli/provider/cmd.go

```diff
@@ -456,15 +475,105 @@ func (c *CmdConfigurator) ValidateFlags(ctx context.Context, cmd *cobra.Command)
 	c.logger.Debug("config has been initialised", zap.Any("for cmd", cmd.Name()), zap.Any("config", c.cfg))
 
 	switch cmd.Name() {
+	case "generate", "download":
+		path, err := cmd.Flags().GetString("path")
+		if err != nil {
+			errMsg := "failed to get the path"
+			utils.LogError(c.logger, err, errMsg)
+			return errors.New(errMsg)
+		}
+
+		c.cfg.Contract.Path = utils.ToAbsPath(c.logger, path)
+
+		services, err := cmd.Flags().GetStringSlice("services")
+		if err != nil {
+			errMsg := "failed to get the services"
+			utils.LogError(c.logger, err, errMsg)
+			return errors.New(errMsg)
+		}
+		config.SetSelectedServices(c.cfg, services)
+
+		selectedTests, err := cmd.Flags().GetStringSlice("tests")
+		if err != nil {
+			errMsg := "failed to get the tests"
+			utils.LogError(c.logger, err, errMsg)
+			return errors.New(errMsg)
+		}
+		config.SetSelectedContractTests(c.cfg, selectedTests)
+
+		if cmd.Name() == "download" {
+			c.cfg.Contract.Driven, err = cmd.Flags().GetString("driven")
+			if err != nil {
+				errMsg := "failed to get the driven flag"
+				utils.LogError(c.logger, err, errMsg)
+				return errors.New(errMsg)
+			}
+
+		}
+
+		c.cfg.Path = utils.ToAbsPath(c.logger, "./")
```

Comment:
Why did you not pass the path variable here?

Reply:
in case of user providing the path for example --path "/home/ahmed/Desktop/"
the c.cfg.path will be  "/home/ahmed/Desktop//keploy" which will be incorrect

Reply:
I think the function will handle this case. Please check at your end. You don't need to send "./"